clients/src/main/java/org/apache/kafka/common/network/NetworkReceive.java

```diff
@@ -37,20 +44,31 @@ public NetworkReceive(String source, ByteBuffer buffer) {
         this.buffer = buffer;
         this.size = null;
         this.maxSize = UNLIMITED;
+        this.memoryPool = MemoryPool.NONE;
     }
 
     public NetworkReceive(String source) {
         this.source = source;
         this.size = ByteBuffer.allocate(4);
         this.buffer = null;
         this.maxSize = UNLIMITED;
+        this.memoryPool = MemoryPool.NONE;
     }
 
     public NetworkReceive(int maxSize, String source) {
         this.source = source;
         this.size = ByteBuffer.allocate(4);
         this.buffer = null;
         this.maxSize = maxSize;
+        this.memoryPool = MemoryPool.NONE;
+    }
+
+    public NetworkReceive(int maxSize, String source, MemoryPool memoryPool) {
```

Comment:
`ControllerShutdown` only uses `BlockingChannel` if the `inter.broker.protocol.version < 0.9.0.0`, so it's safe to ignore. It's only there to allow rolling upgrades from 0.8.x.

I  haven't checked the other cases, it would indeed be good to know if there's a good reason why they are not using the memory pool.